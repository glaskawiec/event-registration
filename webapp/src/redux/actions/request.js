import { apiUri } from '../../config/default.json';

import {
    REQUEST_CREATE_EVENT_PENDING,
    REQUEST_CREATE_EVENT_SUCCESS,
    REQUEST_CREATE_EVENT_FAILURE,
    REQUEST_CREATE_EVENT_RESET
} from '../types/request'

export const createEventPending = () => ({
    type: REQUEST_CREATE_EVENT_PENDING
})

export const createEventSuccess = () => ({
    type: REQUEST_CREATE_EVENT_SUCCESS
})

export const createEventFailure = () => ({
    type: REQUEST_CREATE_EVENT_FAILURE
})

export const createEventReset = () => ({
    type: REQUEST_CREATE_EVENT_RESET
})

export const createEvent = (event) => {
    return dispatch => {
        dispatch(createEventPending());
        return fetch(`${apiUri}/events`, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(event)
        })
            .then((response) => {
                if (response.ok) {
                    dispatch(createEventSuccess())
                } else {
                    dispatch(createEventFailure())
                }
            })
            .catch(() => { dispatch(createEventFailure()) });
    }
}

