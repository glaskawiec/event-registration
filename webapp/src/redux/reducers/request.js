import {
    REQUEST_CREATE_EVENT_PENDING,
    REQUEST_CREATE_EVENT_SUCCESS,
    REQUEST_CREATE_EVENT_FAILURE,
    REQUEST_CREATE_EVENT_RESET
} from '../types/request'

export const initialState = {
    createEvent: {
        isPending: false,
        hadError: false,
        isSucceeded: false,
    }
}

const requestReducer = (state = initialState, action) => {
    switch (action.type) {
        case REQUEST_CREATE_EVENT_PENDING:
            return {
                ...state,
                createEvent: {
                    ...state.createEvent,
                    isPending: true
                }
            }
        case REQUEST_CREATE_EVENT_SUCCESS:
            return {
                ...state,
                createEvent: {
                    ...state.createEvent,
                    isSucceeded: true,
                    isPending: false
                }
            }
        case REQUEST_CREATE_EVENT_FAILURE:
            return {
                ...state,
                createEvent: {
                    ...state.createEvent,
                    isPending: false,
                    hadError: true
                }
            }
        case REQUEST_CREATE_EVENT_RESET:
            return { ...initialState }
        default:
            return state
    }
}

export default requestReducer;