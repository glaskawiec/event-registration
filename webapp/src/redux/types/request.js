export const REQUEST_CREATE_EVENT_PENDING = 'REQUEST_CREATE_EVENT_PENDING'
export const REQUEST_CREATE_EVENT_SUCCESS = 'REQUEST_CREATE_EVENT_SUCCESS'
export const REQUEST_CREATE_EVENT_FAILURE = 'REQUEST_CREATE_EVENT_FAILURE'
export const REQUEST_CREATE_EVENT_RESET = 'REQUEST_CREATE_EVENT_RESET'