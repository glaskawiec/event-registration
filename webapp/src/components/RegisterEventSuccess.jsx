import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';
import { createEventReset } from '../redux/actions/request';
import { connect } from 'react-redux';

export const RegisterEventSuccess = (props) => {
    return (
        <div style={{ textAlign: 'center' }}>
            <h4>Success</h4>
            <p>The data have been saved in the database.</p>
            <Button onClick={props.createEventReset}>Register another</Button>
        </div>
    );
};

RegisterEventSuccess.propTypes = {
    createEventReset: PropTypes.func.isRequired,
}

const ConnectedRegisterEventSuccess = connect(
    () => ({}),
    dispatch => ({
        createEventReset: () => dispatch(createEventReset())
    })
)(RegisterEventSuccess)

export default ConnectedRegisterEventSuccess;

