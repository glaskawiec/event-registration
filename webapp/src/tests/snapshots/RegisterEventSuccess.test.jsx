import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import RegisterEventSuccess from '../../components/RegisterEventSuccess';
import { initialState as requestInitialState } from '../../redux/reducers/request';

const middlewares = []
const mockStore = configureStore(middlewares);

test('RegisterEventSuccess snapshot', () => {
    const initialState = {
        request: requestInitialState
    }
    const store = mockStore(initialState)
    const tree = renderer
        .create(
            <Provider store={store}>
                <RegisterEventSuccess />
            </Provider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});