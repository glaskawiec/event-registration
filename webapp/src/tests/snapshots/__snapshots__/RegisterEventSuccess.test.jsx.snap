// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterEventSuccess snapshot 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h4>
    Success
  </h4>
  <p>
    The data have been saved in the database.
  </p>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Register another
  </button>
</div>
`;
